src/main/java/com/unstoppabledomains/resolution/Resolution.java

```diff
@@ -135,6 +135,16 @@ public String getOwner(String domain) throws NamingServiceException {
         return service.getOwner(domain);
     }
 
+    @Override
+    public String[] getBatchOwners(String[] domains) throws NamingServiceException {
+        for (int i = 0; i + 2 < domains.length; i++) {
+            if (findService(domains[i+1]) != findService(domains[i])) {
+                throw new NamingServiceException(NSExceptionCode.InconsistentDomainArray);
+            }
```

Comment:
Can we use `List` instead of arrays everywhere? 
With a `List` here, you can just do:
```suggestion
           NamingService service = findService(domains.get(0));
           boolean inconsistentDomainArray = domains.stream()
                                                                                       .allMatch(d-> (findService(d) == service));
```